include/MemoryPool.h

```diff
@@ -0,0 +1,35 @@
+#ifndef MEMORY_POOL_H
+#define MEMORY_POOL_H
+
+#include <stdint.h>
+
+/**
+ * A memory pool linked list used to hold and store all the memory that is dynamically allocated
+ * during the life of the program. At the end of the program the user will destroy the memory pool.
+ * 
+ * This is a naive first attempt to have some sort of central memory management.
+ */
+typedef struct MemoryPoolListNode
```

Comment:
I'm wondering if Linked List is the right answer here. I think its fine for now, and the way we are using it, its a write-only data structure for the lifetime of the program, except when we exit, but linked lists are bad for cache locality, so if we wanted to say provide an API to free a specific memory block, this code would quickly get complicated

Reply:
It probably won't be in the long run. I thought it was simply the easiest to have for now.